week4/MealsharingNET/IReservationRepository.cs

```diff
@@ -0,0 +1,12 @@
+using MealsharingNET.Models;
+
+namespace MealsharingNET;
+
+public interface IReservationRepository
+{
+    IEnumerable<Reservation> ListReservations();
+    void Add(Reservation r);
+
+    IEnumerable<Reservation> GetReservation(int id);
```

Comment:
Since you are passing a single id, then the return type should be Reservation. 
Returning an IEnumerable of Reservation would mean that you can have more reservations with the same id.